modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/display/view.jsp

```diff
@@ -46,10 +46,19 @@ request.setAttribute(KBWebKeys.KNOWLEDGE_BASE_KB_NAVIGATION_DISPLAY_CONTEXT, kbN
 
 		<%
 		renderRequest.setAttribute(KBWebKeys.PORTLET_CONFIGURATOR_VISIBILITY, Boolean.TRUE);
+
+		String messageText = null;
+
+		if (KBFolderConstants.DEFAULT_PARENT_FOLDER_ID != kbDisplayPortletInstanceConfiguration.resourcePrimKey()) {
+			messageText = "the-selected-knowledge-base-is-empty";
+		}
+		else {
+			messageText = "please-configure-this-portlet-to-make-it-visible-to-all-users";
+		}
```

Comment:
Can you add this logic as a new `getAlertMessageLabel()` method in the KBNavigationDisplayContext class?

Reply:
Thanks for your feedback @adolfopa. I fixed it.